tests/integration_test/statistics_test.go

```diff
@@ -0,0 +1,138 @@
+				"http.ingress_http.service_control.request_time":  {1000, 1025, 1050, 1075, 1090, 1095, 1099, 1099.5, 1099.9, 1100},
+			},
+		},
+		{
+			desc:        "backend respond in 2s",
+			reqDuration: time.Second * 2,
+			wantCounts: map[string]int{
+				"http.ingress_http.backend_auth.token_added":                       2,
+				"http.ingress_http.backend_routing.append_path_to_address_request": 2,
+				"http.ingress_http.path_matcher.allowed":                           2,
+				"http.ingress_http.service_control.allowed":                        2,
+			},
+			wantHistograms: map[string][]float64{
+				// For overhead_time, only verify the maximum(0 will be ignored).
+				"http.ingress_http.service_control.overhead_time": {0, 0, 0, 0, 0, 0, 0, 0, 0, 5},
+				"http.ingress_http.service_control.backend_time":  {1000, 1050, 1100, 2050, 2080, 2090, 2098, 2099, 2099.8, 2100},
+				"http.ingress_http.service_control.request_time":  {1000, 1050, 1100, 2050, 2080, 2090, 2098, 2099, 2099.8, 2100},
+			},
+		},
+	}
+
+	for _, tc := range testData {
+		path := fmt.Sprintf("/sleepDefault?duration=%v", tc.reqDuration.String())
+		url := fmt.Sprintf("http://localhost:%v%v", s.Ports().ListenerPort, path)
+
+		if _, err := client.DoWithHeaders(url, "GET", "", nil); err != nil {
+			t.Fatalf("Test (%s): failed, expected no err, got err (%v)", tc.desc, err)
+		}
+
+		// Ensure the stats is available in admin.
+		time.Sleep(time.Millisecond * 5000)
+
+		statsUrl := fmt.Sprintf("http://localhost:%v%v", s.Ports().AdminPort, utils.ESpv2FiltersStatsPath())
+		resp, err := utils.DoWithHeaders(statsUrl, "GET", "", nil)
+		if err != nil {
+			t.Fatalf("GET %s faild: %v", url, err)
+		}
+
+		counts, histograms, err := utils.ParseStats(resp)
+		if err != nil {
+			t.Fatalf("fail to parse stats: %v", err)
+		}
+
+		for wantCountName, wantCountVal := range tc.wantCounts {
+			if getCountVal, ok := counts[wantCountName]; !ok {
+				t.Errorf("Test (%s): failed, expected counter %v not in the got counters: %v", tc.desc, wantCountName, counts)
+				break
+			} else if wantCountVal != getCountVal {
+				t.Errorf("Test (%s): failed, for counter %s, expected value %v:, got value: %v ", tc.desc, wantCountName, wantCountVal, getCountVal)
+				break
+			}
+		}
+
+		for wantHistogramName, wantHistogramVals := range tc.wantHistograms {
+			if getHistogramVals, ok := histograms[wantHistogramName]; !ok {
+				t.Errorf("Test (%s): failed, expected histogram %v not in the got histograms: %v", tc.desc, wantHistogramName, histograms)
+				break
+			} else if len(wantHistogramVals) != len(getHistogramVals) {
+				t.Errorf("Test (%s): failed, different value number for histogram %v, expected vals: %v , got vals: %v", tc.desc, wantHistogramName, wantHistogramVals, getHistogramVals)
+			} else {
+				for i, wantHistogramVal := range wantHistogramVals {
+					if wantHistogramVal == 0 {
+						continue
+					}
+					if !roughEqual(getHistogramVals[i], wantHistogramVal, bufferPercent) {
```

Comment:
Not quite understand what we are comparing with.   If we are comparing the latency in ms,  it could be very flaky. very hard to control "bufferPercent"

I think just we just need to make sure number of buckets exist with non-zero data,   not need to actual check the data.

Just like Report,  we don't verify latency values in the report. It can vary dramatically.

Reply:
We can still do comparison here because we make the server delay response at least 1s to make the base very large and with  `bufferPercent` it can mitigate the overhead variation mainly introduced by our proxy, like under different build mode.

Reply:
SG